tests/ui/structs/method-chain-expression-failure.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0599]: no method named `foo` found for struct `D` in the current scope
+  --> $DIR/method-chain-expression-failure.rs:24:19
+   |
+LL | struct D;
+   | -------- method `foo` not found for this struct
+...
+LL |     A.b().c().d().foo();
+   |           ---     ^^^ method not found in `D`
+   |           |
+   |           method `foo` is available on `&B`
```

Comment:
To drive the point home, if you look at the impl the span label should be pointing at the `.b()` which returns `B` and not at the method call `.c()` which returns `C`.